packages/iota/src/validation/outputs/outputs.ts

```diff
@@ -110,9 +122,100 @@ export function validateBasicOutput(object: IBasicOutput, protocolInfo: INodeInf
     }
 
     return mergeValidationResults(...results);
 }
 
+/**
+ * Validate an alias output.
+ * @param aliasOutput The object to validate.
+ * @param protocolInfo The Protocol Info.
+ * @returns The validation result.
+ */
+export function validateAliasOutput(aliasOutput: IAliasOutput, protocolInfo: INodeInfoProtocol): IValidationResult {
+    const results: IValidationResult[] = [];
+
+    if (aliasOutput.type !== ALIAS_OUTPUT_TYPE) {
+        results.push({
+            isValid: false,
+            errors: [`Type mismatch in alias output ${aliasOutput.type}`]
+        });
+    }
+
+    if (bigInt(aliasOutput.amount).compare(bigInt.zero) !== 1) {
+        results.push({
+            isValid: false,
+            errors: ["Alias output amount field must be larger than zero."]
+        });
+    }
+
+    if (bigInt(aliasOutput.amount).compare(protocolInfo.tokenSupply) === 1) {
+        results.push({
+            isValid: false,
+            errors: ["Alias output amount field must not be larger than max token supply."]
+        });
+    }
+
+    if (aliasOutput.unlockConditions.length !== MAX_ALIAS_UNLOCK_CONDITIONS_COUNT) {
+        results.push({
+            isValid: false,
+            errors: [`Unlock conditions count must be equal to ${MAX_ALIAS_UNLOCK_CONDITIONS_COUNT}.`]
+        });
+    }
+
+    if (aliasOutput.unlockConditions) {
+        if (!aliasOutput.unlockConditions.some(uC =>
+            uC.type === STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE ||
+            uC.type === GOVERNOR_ADDRESS_UNLOCK_CONDITION_TYPE
+        )) {
+            results.push({
+                isValid: false,
+                errors: ["State controller address unlock condition or Governor address unlock condition must be present."]
+            });
+        }
+
+        aliasOutput.unlockConditions.map(unlockCondition => {
+            if ((unlockCondition.type === STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE ||
+                unlockCondition.type === GOVERNOR_ADDRESS_UNLOCK_CONDITION_TYPE) &&
+                (unlockCondition.address.type === ALIAS_ADDRESS_TYPE &&
+                    unlockCondition.address.aliasId === aliasOutput.aliasId)) {
+                results.push({
+                    isValid: false,
+                    errors: ["Address of State controller address unlock condition and address of Governor address unlock condition must be different from the Alias address derived from alias id."]
+                });
+            }
+        });
+
+        results.push(validateUnlockConditions(aliasOutput.unlockConditions, aliasOutput.amount, protocolInfo.rentStructure));
+    }
+
+    if (aliasOutput.nativeTokens) {
+        results.push(validateNativeTokens(aliasOutput.nativeTokens));
+    }
+
+    if (aliasOutput.features) {
+        results.push(validateFeatures(aliasOutput.features, MAX_ALIAS_FEATURES_COUNT));
+    }
+
+    if (aliasOutput.immutableFeatures) {
```

Comment:
Feature Type of a Feature in Immutable Features must define on of the following types:

- Issuer Feature

- Metadata Feature

Reply:
Check added.